specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/AppServicePlans.json

```diff
@@ -279,7 +279,7 @@
             "type": "string"
           },
           {
-            "name": "appServicePlan",
+            "name": "appServicePlanPatch",
```

Comment:
the change of "name" on body parameter seems unnecessary, as probably only python uses the "name" (it won't affect REST API, but it may break SDK if they indeed uses the "name" of body).

Reply:
@jackofallops FYI

Reply:
This makes the SDK generation non-deterministic? If this is used as is, the wrong model will be selected (without some override) in some languages SDKs (Python/Go etc). So the PATCH will use the PUT model here, so this is correcting this. This should have a unique name in this spec as it refers to a different model to the PUT, as @JeffreyRichter called out above:

>That being said, if the service has already GA'd then we don't want to introduce a breaking change. So is this change correcting the swagger to make it accurately reflect the currently-shipping service behavior?

Ideally, we'd not want this to refer to another model, as this doesn't match standard behaviour in this and other APIs, but if there's legitimate reason not to, we should update here for the name to be unique?

Reply:
This is a parameter name, not a model name.

The model is here (which is the correct one `AppServicePlanPatchResource`?)
https://github.com/Azure/azure-rest-api-specs/blob/f0e0f76639744ed797a1811267012505ab62a947/specification/web/resource-manager/Microsoft.Web/stable/2022-09-01/AppServicePlans.json#L286-L288

This model will be used whatever the parameter name.

Reply:
@jackofallops Could you please let us know the intended SDK ? Depending on that we can tag the right SDK owners.

Reply:
@navba-MSFT this change is for github.com/hashicorp/go-azure-sdk (so we'd be the owners/folks to tag there, fwiw)

Reply:
It will not break Azure Go SDK. But it is labeled as a swagger breaking change (though I don't think it will break payload), it's better to get approval from Jeff.

Reply:
Here's what I think is happening: Today, there is an SDK-only method parameter named appServicePlan and there is desire to change this name to appServicePlanPatch. I'm not sure why this desire exists but the only real language it might affect is C#; not Go. And this has such a small chance of breaking .NET that we'd approve it. But I don't understand the motivation for making this change? 

Do I have this right?